docs/source/develop.rst

```diff
@@ -79,17 +79,24 @@ Make a fork of the main `Dask repository <https://github.com/dask/dask>`_ and
 clone the fork::
 
    git clone https://github.com/<your-github-username>/dask
+   cd dask
+
+You should also pull the latest tags (otherwise the pip dependency resolver can
+sometimes get stuck installing Dask)::
+
+   git remote add upstream https://github.com/dask/dask
+   git pull upstream --tags
 
 Contributions to Dask can then be made by submitting pull requests on GitHub.
 
 
 Install
 ~~~~~~~
 
-To build the library you can install the necessary requirements using
-pip or conda_::
+First, make sure you are in the top level of your cloned Dask repository.
 
-  cd dask
+To build the library you can install the necessary requirements using
+pip or conda_
```

Comment:
I think we can probably consolidate these two sentences

> From the top level of your cloned Dask repository you can install a
local version of Dask, along with all necessary dependencies, using
pip or conda_.